src/migrations/20241119105837-licensed-users-backfill.js

```diff
@@ -0,0 +1,85 @@
+exports.up = (db, cb) => {
+  db.runSql(`
+        WITH user_events AS (
+          SELECT
+              DISTINCT CASE WHEN type = 'user-deleted' THEN pre_data ->> 'email' ELSE data ->> 'email' END AS email,
+                       type,
+                       created_at AS event_date
+          FROM
+              events
+          WHERE
+              type IN ('user-created', 'user-deleted')
+        ),
+        dates AS (
+           WITH RECURSIVE generated_dates AS (
+               SELECT
+                   MIN(event_date):: timestamp AS date
+               FROM
+                   user_events
+               UNION ALL
+               SELECT
+                   date + INTERVAL '1 day'
+               FROM
+                   generated_dates
+               WHERE
+                   date + INTERVAL '1 day' <= CURRENT_DATE
+           )
+           SELECT
+               date :: date
+           FROM
+               generated_dates
+        ),
+        active_emails AS (
+           SELECT
+               d.date,
+               ue.email,
+               MAX(
+                   CASE WHEN ue.type = 'user-created' THEN ue.event_date ELSE NULL END
+               ) AS created_date,
+               MAX(
+                   CASE WHEN ue.type = 'user-deleted' THEN ue.event_date ELSE NULL END
+               ) AS deleted_date
+           FROM
+               dates d
+                   LEFT JOIN user_events ue ON ue.event_date <= d.date
+           GROUP BY
+               d.date,
+               ue.email
+        ),
+        result AS (
```

Comment:
Count the emails on date, 1 if the deleted date is empty, delete happened within last 30 days, delete happened before the creation.